### PR TITLE
Better delay description

### PR DESCRIPTION
**Pull request by therealprof**
CC #236

Signed-off-by: Daniel Egger <daniel@eggers-club.de>

### CODE REVIEW

File: src/asm.rs [hunk 1]
```diff
@@ -26,13 +26,16 @@ pub fn bkpt() {
...
 /// Blocks the program for *at least* `n` instruction cycles
 ///
-/// This is implemented in assembly so its execution time is the same regardless of the optimization
-/// level.
+/// This is implemented in assembly so its execution time is independent of the optimization
+/// level, however it is dependent on the specific architecture and core configuration.
 ///
-/// NOTE that the delay can take much longer if interrupts are serviced during its execution.
+/// NOTE that the delay can take much longer if interrupts are serviced during its execution 
+/// and the execution time may wary with other factors. This delay is mainly useful for simple
```
Comment (jonas-schievink) [inline src/asm.rs:33]:
```suggestion
/// and the execution time may vary with other factors. This delay is mainly useful for simple
```

File: src/asm.rs [hunk 2]
```diff
@@ -26,13 +26,16 @@ pub fn bkpt() {
...
+/// timer-less initialisation of peripherals iff accurate timing is not essential. In any other
```
Comment (jonas-schievink) [inline src/asm.rs:34]:
```suggestion
/// timer-less initialization of peripherals if accurate timing is not essential. In any other
```

Comment (therealprof) [reply src/asm.rs:34]: Hm, the **iff** is intended. I'm fine with changing the spelling to AE.

Comment (jonas-schievink) [reply src/asm.rs]: Maybe expand it to "if and only if" then? *iff* tends to confuse people.

Comment (therealprof) [reply src/asm.rs]: Hm, I've never met anyone who was confused about that but sure, why not.

**Pull request merged**